dali/test/tensor_adapter.h

```diff
@@ -0,0 +1,36 @@
+#ifndef DALI_TENSOR_BATCH_ADAPTER_H
+#define DALI_TENSOR_BATCH_ADAPTER_H
+
+#include "../pipeline/data/tensor_list.h"
+#include <vector>
+
+namespace dali {
+
+namespace testing {
+
+template<typename UnderlyingDataType>
+class TensorAdapter {
```

Comment:
This class doesn't provide the conversion facilities we've talked about.
It leaves the burden of placing reference data in contiguous blocks on the test developer.
It hides the Tensor object from the developer _and_ is used for verification - the thing we've agreed (?) is not what we really want.

Reply:
As I imagined it, test developer creates a vector of `TensorAdapter`s. Passes it to DaliOperatorTest using GenerateInputs and underneath `dali::TensorList` is created using `ToTensorList`. And this is the place, where we care about contiguous mem block, which can be done automatically. Is it ok? Or I missed something important?